great_expectations/datasource/data_connector/configured_asset_sql_data_connector.py

```diff
@@ -173,14 +177,29 @@ def get_unmatched_data_references(self) -> List[str]:
         return []
 
     def get_batch_definition_list_from_batch_request(self, batch_request: BatchRequest):
+        """
+        Retrieve batch_definitions that match batch_request
+
+        First retrieves all batch_definitions that match batch_request
+            - if batch_request also has a batch_filter, then select batch_definitions that match batch_filter.
+            - NOTE : currently sql data connectors do not support sorters.
+
+        Args:
+            batch_request (BatchRequestBase): BatchRequestBase (BatchRequest without attribute validation) to process
+
+        Returns:
+            A list of BatchDefinition objects that match BatchRequest
+        """
         self._validate_batch_request(batch_request=batch_request)
 
         if len(self._data_references_cache) == 0:
             self._refresh_data_references_cache()
 
         batch_definition_list: List[BatchDefinition] = []
         try:
-            sub_cache = self._data_references_cache[batch_request.data_asset_name]
+            sub_cache = self._get_data_reference_list_from_cache_by_data_asset_name(
+                data_asset_name=batch_request.data_asset_name
+            )
```

Comment:
@Shinnnyshinshin Small issue -- we may have a "use before create" situation with `sub_cache`.  Can be easily resolved by setting it at `None` initially.  Thanks.